firefox_puppeteer/api/places.py

```diff
@@ -77,30 +77,29 @@ def is_bookmark_star_button_ready(self):
 
     def restore_default_bookmarks(self):
         """Restores the default bookmarks for the current profile."""
-        try:
-            return self.marionette.execute_async_script("""
-              Cu.import("resource://gre/modules/BookmarkHTMLUtils.jsm");
-              Cu.import("resource://gre/modules/Services.jsm");
-
-              // Default bookmarks.html file is stored inside omni.jar,
-              // so get it via a resource URI
-              let defaultBookmarks = 'resource:///defaults/profile/bookmarks.html';
-
-              let observer = {
+        retVal = self.marionette.execute_async_script("""
+            Cu.import("resource://gre/modules/BookmarkHTMLUtils.jsm");
+            Cu.import("resource://gre/modules/Services.jsm");
+            // Default bookmarks.html file is stored inside omni.jar,
+            // so get it via a resource URI
+            let defaultBookmarks = 'resource:///defaults/profile/bookmarks.html';
+            let observer = {
                 observe: function (aSubject, aTopic, aData) {
-                  Services.obs.removeObserver(observer, "bookmarks-restore-success");
+                Services.obs.removeObserver(observer, "bookmarks-restore-failed");
+                Services.obs.removeObserver(observer, "bookmarks-restore-success");
 
-                  marionetteScriptFinished(true);
+                marionetteScriptFinished(aTopic == "bookmarks-restore-success");
                 }
-              };
-
-              // Trigger the import of the default bookmarks
-              Services.obs.addObserver(observer, "bookmarks-restore-success", false);
-              BookmarkHTMLUtils.importFromURL(defaultBookmarks, true);
-            """, script_timeout=10000)
-        except TimeoutException:
-            # TODO: In case of a timeout clean-up the registered topic
-            pass
+            };
+            // Trigger the import of the default bookmarks
+            Services.obs.addObserver(observer, "bookmarks-restore-failed", false);
+            Services.obs.addObserver(observer, "bookmarks-restore-success", false);
+            BookmarkHTMLUtils.importFromURL(defaultBookmarks, true);
+        """, script_timeout=10000)
+
+        if not retVal:
+            raise Exception("Restore Default Bookmarks failed")
+        return retVal
```

Comment:
I don't think there is a need to return this value here. We already raise an exception in case of a failed restoration. So we should return nothing. Just remove this line.